torch/csrc/jit/script/compiler.cpp

```diff
@@ -260,97 +263,272 @@ at::Tensor getAttributeValue(const NamedValue& nv) {
-    if(first->type()->kind() != TupleType::Kind) {
-      throw ErrorReport(loc) << "expected a tuple";
+  // check input types
+  std::vector<Value*> flat_inputs;
+  for(size_t i = 0; i < schema.arguments.size(); ++i) {
+    NamedValue v = *positional_inputs[i];
+    const auto& arg = schema.arguments[i];
+
+    // implicit conversion from List[Tensor] -> Tensor for when the argument
+    // is an IntList in aten, for things like x.expand(sizes=[3,4,5])
+    if(arg.attribute_kind == AttributeKind::is &&
+       v.value->type()->isSubtypeOf(*ListType::ofTensors())) {
+      auto unpacked = graph->insertNode(graph->createTupleUnpack(v.value));
+      v.value = createStack(*graph, loc, unpacked->outputs());
     }
 
-    if(attributes.size() == 1) {
-      if(inputs.size() > 1) {
-        throw ErrorReport(loc) << "expected 1 input";
-      }
+    if(!v.value->type()->isSubtypeOf(*arg.type)) {
+      err() << "expected a value of type " << arg.type->name() << " for argument '" << arg.name << "' but found "
+            << v.value->type()->name() << "\n"
+            << v.loc;
+      return nullptr;
+    }
+
+    // we only support lists for builtins, where they must be flattened
+    if(arg.type->kind() == TypeKind::ListType) {
+      auto unpacked = graph->insertNode(graph->createTupleUnpack(v.value));
+      const auto & outputs = unpacked->outputs();
+      flat_inputs.insert(flat_inputs.end(), outputs.begin(), outputs.end());
     } else {
-      // findTensorOp already verified we don't have additional attributes
-      JIT_ASSERT(attributes.size() == 0);
-      if(inputs.size() != 2) {
-          throw ErrorReport(loc) << "expected 2 inputs";
-      }
+      flat_inputs.push_back(v.value);
     }
 
-    // flatten the tuple into the argument list
-    auto unpacked = graph->insertNode(graph->createTupleUnpack(first));
-    ensureTensors(loc, unpacked->outputs());
-    n->removeInput(0);
-    for(size_t i = 0; i < unpacked->outputs().size(); ++i) {
-      n->insertInput(i, unpacked->outputs().at(i));
+  }
+
+  // we successfully matched this schema, construct the node
+
+  // note: we always construct a purely positional nodes here
+  // the pass liftConstantAttributes replaces the node with with one that
+  // uses attributes if all the attributes ended up as constants
+
+  NodeKind kind(Symbol::aten(name));
+  auto n = graph->insertNode(graph->create(kind, flat_inputs, 0))
+                ->setSourceLocation(std::make_shared<SourceRange>(loc));
+
+  size_t num_outputs = schema.returns.size();
+
+  // special case for chunk when the chunks=<const> is known
+  // DO NOT ADD MORE SPECIAL CASES HERE, REFACTOR INTO A FUNCTION IF
+  // NEEDED
+  if(n->kind() == aten::chunk) {
+    auto value = constant_as<int64_t>(flat_inputs[1]);
+    if(!value) {
+      throw ErrorReport(positional_inputs[1]->loc) << "argument 'chunks' must be a constant";
     }
-  } else {
-    ensureTensors(loc, inputs);
+    num_outputs = *value;
```

Comment:
Actually what's `num_outputs` for chunk?

Reply:
It is equal to the number of chunks specified in the chunks=argument. This is why we are checking that this argument is constant here.

Reply:
I was curious what's in that field in the schema